### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by anitorosyanvolo**

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+  let fixture: ComponentFixture<AppComponent>;
+  let app: AppComponent;
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [AppComponent],
     }).compileComponents();
+    fixture = TestBed.createComponent(AppComponent);
+    app = fixture.componentInstance;
+  });
+  it('should do something', () => {
```
Comment (Armenvardanyan95) [inline src/app/app.component.spec.ts:15]: It really does something 😁

File: src/app/app.component.spec.ts [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+    const app = fixture.componentInstance;
+    expect(app).toBeTruthy();
+  });
+  it('should render router-outlet', () => {
+    expect(fixture.debugElement.query(By.css('router-outlet'))).toBeTruthy();
```
Comment (Armenvardanyan95) [inline src/app/app.component.spec.ts:20]: Thanks!

File: src/app/add-transaction.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }).compileComponents();
+    fixture = TestBed.createComponent(AddExpenseComponent);
+    component = fixture.componentInstance;
+  });
+  it('should hide modal in cancel', () => {
+     component.cancel()
+     expect(component.open()).toBe(false);
+     expect(component.type()).toBe('expense');
+  });
+  it('should test addTransaction', () => {
```
Comment (Armenvardanyan95) [inline src/app/add-transaction.component.spec.ts:19]: Could have been more explicit here, like "should not submit when invalid and submit the valid amount"

File: src/app/add-transaction.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    component.addTransaction();
+    expect(resetSpy).toHaveBeenCalled();
+    expect(transactionAddedSpy).toHaveBeenCalled();
+    component.type.set('income');
+    component.form.patchValue({
+        amount: 20,
+        title: 'Test1',
+    });
+    component.addTransaction();
+    expect(component.form.controls.amount.value).toBe(20);
```
Comment (Armenvardanyan95) [inline src/app/add-transaction.component.spec.ts:37]: I would add a test for negative numbers

File: src/app/transactions.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    let component: TransactionsComponent;
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [TransactionsComponent],
+    }).compileComponents();
+    fixture = TestBed.createComponent(TransactionsComponent);
+    component = fixture.componentInstance;
+  });
+  it('should open dialog', () => {
+    component.openAddTransactionDialog();
```
Comment (Armenvardanyan95) [inline src/app/transactions.component.spec.ts:16]: Test the event that triggers it

File: src/app/transactions.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    expect(fixture.debugElement.query(By.css('table')))
+    const children = fixture.debugElement.query(By.css('table')).queryAll(By.css('tr'));
+    const length = component.transactions().length;
+    component.deleteTransaction(1);
+    fixture.detectChanges();
+    expect(fixture.debugElement.query(By.css('table')).queryAll(By.css('tr')).length).toBe(children.length - 1);
+    expect(component.transactions().length).toBe(length-1);
+  });
+  it('should check currentBalance', () => {
+    const balance = component.currentBalance();
```
Comment (Armenvardanyan95) [inline src/app/transactions.component.spec.ts:30]: I would check the correct initial/updated `currentBalance` in the DOM

**Review by Armenvardanyan95 — COMMENTED** [review]
Not bad in general, but should add way more tests that interact with the DOM

**Review by Armenvardanyan95 — APPROVED** [review]
